drivers/i2c/i2c_mspm0.c

```diff
@@ -0,0 +1,859 @@
+/*
+ * Copyright (c) 2024 Texas Instruments
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT ti_mspm0_i2c
+
+/* Zephyr includes */
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/drivers/clock_control.h>
+#include <zephyr/drivers/clock_control/mspm0_clock_control.h>
+#include <soc.h>
+
+/* Logging includes */
+#define LOG_LEVEL CONFIG_I2C_LOG_LEVEL
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(i2c_mspm0);
+#include "i2c-priv.h"
+
+/* Driverlib includes */
+#include <ti/driverlib/dl_i2c.h>
+#include <ti/driverlib/dl_gpio.h>
+
+#define TI_MSPM0_TARGET_TIMEOUT_50_MS (193)
+
+#define TI_MSPM0_CONTROLLER_INTERRUPTS	\
+	(DL_I2C_INTERRUPT_CONTROLLER_ARBITRATION_LOST |	\
+	DL_I2C_INTERRUPT_CONTROLLER_NACK |	\
+	DL_I2C_INTERRUPT_CONTROLLER_RXFIFO_TRIGGER |	\
+	DL_I2C_INTERRUPT_CONTROLLER_RX_DONE | DL_I2C_INTERRUPT_CONTROLLER_TX_DONE \
+	IF_ENABLED(CONFIG_I2C_CONTROLLER_TIMEOUT, ( | DL_I2C_INTERRUPT_TIMEOUT_A)))
+
+#define TI_MSPM0_TARGET_INTERRUPTS	\
+	(DL_I2C_INTERRUPT_TARGET_RX_DONE | DL_I2C_INTERRUPT_TARGET_TXFIFO_EMPTY |	\
+	DL_I2C_INTERRUPT_TARGET_START |	DL_I2C_INTERRUPT_TARGET_STOP	\
+	IF_ENABLED(CONFIG_I2C_TARGET_TIMEOUT, ( | DL_I2C_INTERRUPT_TIMEOUT_A)))
+
+enum i2c_mspm0_state {
+	I2C_MSPM0_IDLE,
+	I2C_MSPM0_TX_STARTED,
+	I2C_MSPM0_TX_INPROGRESS,
+	I2C_MSPM0_TX_COMPLETE,
+	I2C_MSPM0_RX_STARTED,
+	I2C_MSPM0_RX_INPROGRESS,
+	I2C_MSPM0_RX_COMPLETE,
+	I2C_MSPM0_TARGET_STARTED,
+	I2C_MSPM0_TARGET_TX_INPROGRESS,
+	I2C_MSPM0_TARGET_RX_INPROGRESS,
+	I2C_MSPM0_TARGET_PREEMPTED,
+	I2C_MSPM0_TIMEOUT,
+	I2C_MSPM0_ERROR
+};
+
+struct i2c_mspm0_config {
+	I2C_Regs *base;
+	uint32_t bitrate;
+	DL_I2C_ClockConfig gI2CClockConfig;
```

Comment:
Zephyr does not allow camel-case style (unless for the SDK types, like here, but the attribute itself should not)

Apply that comment everywhere where relevant (there are many attributes and variables that do not comply)